src/test/scala/org/broadinstitute/dsde/agora/server/webservice/CompatibleConfigurationIntegrationSpec.scala

```diff
@@ -30,80 +30,123 @@ class CompatibleConfigurationIntegrationSpec extends FreeSpec with ExecutionDire
   val methodsService = new MethodsService(permsDataSource) with ActorRefFactoryContext
   val testRoutes = ApiService.handleExceptionsAndRejections (methodsService.queryCompatibleConfigurationsRoute)
 
+  // The following methods create a method AgoraEntity while at the same time constructing the Mock Waas
+  // describe response.  This test is more difficult than the others because it generates many unique
+  // method payloads (WDLs) so it can test the compatibility of methods
+  private def generateMethodWithWaasResponse(label:String, inputs:Seq[String], outputs:Seq[String], optionalInputs:Seq[String] = Seq.empty[String]) = {
+    val method = testMethod(label, inputs, outputs, optionalInputs)
+    addSubstringResponse(method.payload.get, waasResponse(inputs, outputs, optionalInputs))
+    method
+  }
```

Comment:
I'm not gonna hold the PR for this, but for the sake of engineering goodness - this method is side-effecty. It returns the method payload (nice and stateless) but also writes to the mockserver instance (side effect). In an ideal world, this method would either be fully stateless: return a (method, waasResponse) pair that calling code inserts to the right places, OR fully action-oriented: insert the method into the db and set up the mockserver response.

Again, not worth refactoring, especially since this is test code, if it's a big deal and would delay the  PR.

Reply:
done!